lib/manager/components/GeneralSettings.js

```diff
@@ -0,0 +1,202 @@
+// @flow
+
+import Icon from '@conveyal/woonerf/components/icon'
+import React, {Component} from 'react'
+import {
+  Col,
+  ListGroup,
+  ListGroupItem,
+  Button,
+  Panel,
+  FormControl,
+  InputGroup,
+  ControlLabel,
+  FormGroup,
+  Checkbox
+} from 'react-bootstrap'
+
+import * as feedsActions from '../actions/feeds'
+
+import type {Feed, Project} from '../../types'
+import type {ManagerUserState} from '../../types/reducers'
+
+type Props = {
+  confirmDeleteFeedSource: () => void,
+  disabled: ?boolean,
+  feedSource: Feed,
+  project: Project,
+  updateFeedSource: typeof feedsActions.updateFeedSource,
+  user: ManagerUserState
+}
+
+type State = {
+  name?: ?string,
+  url?: ?string
+}
+
+export default class GeneralSettings extends Component<Props, State> {
+  state = {}
+
+  _onChange = ({target}: SyntheticInputEvent<HTMLInputElement>) => {
+    // Change empty string to null to avoid setting URL to empty string value.
+    const value = target.value === '' ? null : target.value.trim()
+    this.setState({[target.name]: value})
+  }
+
+  _onToggleDeployable = () => {
+    const {feedSource, updateFeedSource} = this.props
+    updateFeedSource(feedSource, {deployable: !feedSource.deployable})
+  }
+
+  _getFormValue = (key: 'name' | 'url') => {
+    // If state value does not exist (i.e., form is unedited), revert to value
+    // from props.
+    const value = typeof this.state[key] === 'undefined'
+      ? this.props.feedSource[key]
+      : this.state[key]
+    // Revert to empty string to avoid console error with null value for form.
+    return value || ''
+  }
+
+  _onToggleAutoFetch = () => {
+    const {feedSource, updateFeedSource} = this.props
+    const value = feedSource.retrievalMethod === 'FETCHED_AUTOMATICALLY'
+      ? 'MANUALLY_UPLOADED'
+      : 'FETCHED_AUTOMATICALLY'
+    updateFeedSource(feedSource, {retrievalMethod: value})
+  }
+
+  _onTogglePublic = () => {
+    const {feedSource, updateFeedSource} = this.props
+    updateFeedSource(feedSource, {isPublic: !feedSource.isPublic})
+  }
+
+  _onNameChanged = (evt: SyntheticInputEvent<HTMLInputElement>) => {
+    this.setState({name: evt.target.value})
+  }
+
+  _onNameSaved = () => {
+    const {feedSource, updateFeedSource} = this.props
+    updateFeedSource(feedSource, {name: this.state.name})
+  }
+
+  _onSaveUrl = () => {
+    const {feedSource, updateFeedSource} = this.props
+    updateFeedSource(feedSource, {url: this.state.url})
+  }
+
+  render () {
```

Comment:
Does this need a review or is it basically extracted without modification from the `FeedSourceSettings` component?

Reply:
Yea, exactly. This is basically extracted to make file management a little easier.